frontend/src/metabase/common/components/AccordionList/utils.ts

```diff
@@ -174,3 +176,66 @@ export function isReactNode(x: unknown): x is ReactNode {
     x === undefined
   );
 }
+
+export const getSearchIndex = memoize(function <
+  TItem extends Item,
+  TSection extends Section<TItem>,
+>({
+  sections,
+  searchProp = ["name", "displayName"] as unknown as SearchProps<TItem>,
+}: {
+  sections: TSection[];
+  searchProp?: SearchProps<TItem>;
+}) {
+  const items = sections.flatMap((section) => section.items ?? []);
+  const keys = Array.isArray(searchProp) ? searchProp : [searchProp];
+
+  return new Fuse<TItem>(items, {
+    keys,
+    includeScore: true,
+    isCaseSensitive: false,
+  });
+});
+
+/**
+ * Memoizes a function based on shallow equality of its argument.
+ */
+function memoize<T extends object, R>(fn: (t: T) => R): (t: T) => R {
```

Comment:
Do we need to introduce our own memo function here, or can we use it library that it's already bundled in the app for this?

Reply:
Hasher needs to return a string, so it won't work. We need object identity and there's no way to get that in JavaScript other than directly comparing objects with `===`.